assets/overlays/aws-ebs/aws_ebs.go

```diff
@@ -0,0 +1,72 @@
+package aws_ebs
+
+import (
+	"github.com/openshift/csi-operator/pkg/config/common"
+	"github.com/openshift/csi-operator/pkg/generator"
+)
+
+// GetAWSEBSGeneratorConfig returns configuration for generating assets of  AWS EBS CSI driver operator.
+func GetAWSEBSGeneratorConfig() *generator.CSIDriverGeneratorConfig {
+	return &generator.CSIDriverGeneratorConfig{
+		AssetPrefix:      "aws-ebs-csi-driver",
+		AssetShortPrefix: "ebs",
+		DriverName:       "ebs.csi.aws.com",
+
+		ControllerConfig: &generator.ControlPlaneConfig{
+			DeploymentTemplateAssetName: "overlays/aws-ebs/patches/controller_add_driver.yaml",
+			LivenessProbePort:           10301,
+			MetricsPorts: []generator.MetricsPort{
+				{
+					LocalPort:           common.AWSEBSLoopbackMetricsPortStart,
+					InjectKubeRBACProxy: true,
+					ExposedPort:         common.AWSEBSExposedMetricsPortStart,
+					Name:                "driver-m",
+				},
+			},
+			SidecarLocalMetricsPortStart:   common.AWSEBSLoopbackMetricsPortStart + 1,
+			SidecarExposedMetricsPortStart: common.AWSEBSExposedMetricsPortStart + 1,
+			Sidecars: []generator.SidecarConfig{
+				common.DefaultProvisionerWithSnapshots.WithExtraArguments(
+					"--default-fstype=ext4",
+					"--feature-gates=Topology=true",
+					"--extra-create-metadata=true",
+					"--timeout=60s",
+				),
+				common.DefaultAttacher.WithExtraArguments(
+					"--timeout=60s",
+				),
+				common.DefaultResizer.WithExtraArguments(
+					"--timeout=300s",
+				),
+				common.DefaultSnapshotter.WithExtraArguments(
+					"--timeout=300s",
+					"--extra-create-metadata",
+				),
+				common.DefaultLivenessProbe.WithExtraArguments(
+					"--probe-timeout=3s",
+				),
+			},
+			Assets: common.DefaultControllerAssets,
+			AssetPatches: common.DefaultAssetPatches.WithPatches(generator.HyperShiftOnly,
+				"controller.yaml", "overlays/aws-ebs/patches/controller_add_hypershift_controller_minter.yaml",
```

Comment:
It is bit weird to see use of sets here. So this basically defines - in hypershift environment, for `controller.yaml` apply this patch?

Reply:
I feel like there is an assumption that `ControlPlaneConfig` will produce `controller.yaml`. Can we declare this as a field, which defaults to `controller.yaml` so as user does not have specify it, but it can be used to refer the code and verify what resource this section produces.

Reply:
Yes, exactly. I am not sure there is a patch that's applied to both flavours, but there are _assets_ that are (e.g. storageclass_gp2.yaml in this file) and then there are flavour-specific patches (`common/standalone/controller_add_affinity.yaml` / `common/hypershift/controller_add_affinity_tolerations.yaml`).